packages/design-system-react-native/src/components/Icons/Icon/Icon.tsx

```diff
@@ -0,0 +1,33 @@
+import {
+  useTailwind,
+  withThemeProvider,
+} from '@metamask/design-system-twrnc-preset';
+import React from 'react';
+
+import { assetByIconName } from './Icon.assets';
+import { DEFAULT_ICON_SIZE, DEFAULT_ICON_COLOR } from './Icon.constants';
+import type { IconProps } from './Icon.types';
+import { generateIconClassNames } from './Icon.utilities';
+
+const Icon = ({
+  size = DEFAULT_ICON_SIZE,
+  name,
+  color = DEFAULT_ICON_COLOR,
+  style,
+  ...props
+}: IconProps) => {
+  const tw = useTailwind();
+  const SVG = assetByIconName[name];
+  const mergedClassnames = generateIconClassNames({ color, size });
```

Comment:
I think there's an opportunity to optimize the style generation. Currently, `generateIconClassNames` and the subsequent `tw` transformation are recalculated on every render. We can optimize this by memoizing both the class name generation and the tw transformation.
```suggestion
  const styles = useMemo(() => {
    const mergedClassnames = generateIconClassNames({ color, size });
    return tw`${mergedClassnames}`;
  }, [color, size, tw]);
```

Then in the svg
```
  <SVG
    name={name}
    fill="currentColor"
    style={[styles]}
    {...props}
  />
```